lib/src/executable/watch.dart

```diff
@@ -226,34 +182,41 @@ final class _Watcher {
   /// Returns whether all recompilations succeeded.
   Future<bool> _recompileDownstream(Iterable<StylesheetNode> nodes) async {
     var seen = <StylesheetNode>{};
-    var toRecompile = Queue.of(nodes);
-
     var allSucceeded = true;
-    while (toRecompile.isNotEmpty) {
-      var node = toRecompile.removeFirst();
-      if (!seen.add(node)) continue;
-
-      var success = await _compileIfEntrypoint(node.canonicalUrl);
-      allSucceeded = allSucceeded && success;
-      if (!success && _options.stopOnError) return false;
+    while (nodes.isNotEmpty) {
+      nodes = nodes.where((node) => seen.add(node)).toList();
```

Comment:
It's more idiomatic these days to use list syntax for this sort of thing:

```suggestion
      nodes = [for (var node in nodes) if (seen.add(node)) node];
```

Also below.